reference-architecture/vmware-ansible/roles/haproxy-server/tasks/main.yaml

```diff
@@ -0,0 +1,36 @@
+---
+- name: Ensure HAProxy is installed.
+  yum: name=haproxy state=installed
+  when: inventory_hostname == 'haproxy-0'
+
+- name: Get HAProxy version.
+  command: haproxy -v
+  register: haproxy_version_result
+  changed_when: false
+  always_run: yes
+  when: inventory_hostname == 'haproxy-0'
+
+- name: open firewall for Openshift services
+  command: iptables -I INPUT -p tcp --dport {{item}} -j ACCEPT
+  with_items:
+    - 8443
+    - 443
+    - 80
+  when: inventory_hostname == 'haproxy-0'
+
+- name: Save the iptables rules
+  command: iptables-save
+  when: inventory_hostname == 'haproxy-0'
```

Comment:
The handlers for this role reference firewalld, but you are using iptables commands here. If firewalld is a requirement, I would suggest using the firwalld Ansible module instead.